### PR TITLE
Better debugging for "Your test case is not allowed to access the dat…

### PR DESCRIPTION
**Pull request by nickvergessen**
…abase."

Ref https://help.nextcloud.com/t/integration-test-failure/28383

### before
```
There was 1 error:

1) OCA\FocusFrog\Tests\Integration\Controller\TaskIntegrationTest::testUpdate
OCP\AppFramework\QueryException: Could not resolve service! Class service does not exist

/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:108
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:123
/home/nickv/Nextcloud/13/server/lib/private/ServerContainer.php:132
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:441
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:79
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:102
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:123
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:467
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:438
/home/nickv/Nextcloud/13/server/appstank/focusfrog/tests/Integration/Controller/TaskIntegrationTest.php:30

ERRORS!
Tests: 2, Assertions: 1, Errors: 1.
```

### before
```
There was 1 failure:

1) OCA\FocusFrog\Tests\Integration\Controller\TaskIntegrationTest::testUpdate
Your test case is not allowed to access the database.

/home/nickv/Nextcloud/13/server/tests/lib/TestCase.php:142
/home/nickv/Nextcloud/13/server/3rdparty/pimple/pimple/src/Pimple/Container.php:113
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:123
/home/nickv/Nextcloud/13/server/lib/private/ServerContainer.php:132
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:441
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:68
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:104
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:125
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:467
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:438
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:68
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:104
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:125
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:467
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:438
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:68
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:104
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/Utility/SimpleContainer.php:125
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:467
/home/nickv/Nextcloud/13/server/lib/private/AppFramework/DependencyInjection/DIContainer.php:438
/home/nickv/Nextcloud/13/server/appstank/focusfrog/tests/Integration/Controller/TaskIntegrationTest.php:30

FAILURES!
Tests: 2, Assertions: 1, Failures: 1.

```

### CODE REVIEW

File: lib/private/AppFramework/Utility/SimpleContainer.php [hunk 1]
```diff
@@ -66,6 +66,12 @@ private function buildClass(ReflectionClass $class) {
 				try {
 					$parameters[] = $this->query($resolveName);
 				} catch (\Exception $e) {
+					if (class_exists("PHPUnit_Framework_AssertionFailedError") &&
+						$e instanceof \PHPUnit_Framework_AssertionFailedError) {
```
Comment (nickvergessen) [inline lib/private/AppFramework/Utility/SimpleContainer.php]: don't like this too much, I first tried to do it for `resolveName =IDBConnection  but then it get's lost on nested dependendies...
So this is the only way we can do it.

**Review by MorrisJobke — APPROVED** [review]
I don't fully like it, but it seems to be the only way to accomplish this kind of nested DI container query detection :/

**Review by rullzer — APPROVED** [review]

**Review by rullzer — CHANGES_REQUESTED** [review]
Phan is not happy with this....

**Comment by nickvergessen** [comment]
Undeclared class, yeah that's why it is in class_exists.... Maybe denying autoload helps, but i dont think phan is so clever...

**Comment by MorrisJobke** [comment]
> Phan is not happy with this....

Still not happy :(

lib/private/AppFramework/Utility/SimpleContainer.php:71 PhanUndeclaredClassInstanceof Checking instanceof against undeclared class \PHPUnit_Framework_AssertionFailedError

**Comment by MorrisJobke** [comment]
> Still not happy :(

Maybe put it in one line instead of two?

**Comment by nickvergessen** [comment]
Right, reported at https://github.com/phan/phan/issues/1543

And fixed it by moving the suppress to the method docs.

**Review by MorrisJobke — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Pull request merged**